### ISSUE TITLE
"target_lengths must be of size batch_size" during validation

### ISSUE THREAD

**Issue by ongcj** [novioleo/FOTS#11]
Hi @novioleo ,
for every validation during training, it keeps saying "target_lengths must be of size batch_size". I am trying to train using batch size of 4 using ICDAR2015 data set. Hope you could help me with this issue. Thanks

**Comment by novioleo** [comment]
@ongcj
this warning come from ctc loss.
you can print the loss exception in recognition loss.
and i'll fix these exceptions in some days.
pls be patient

**Comment by novioleo** [comment]
@ongcj 
i think over and over again, i think there is **no nessary** to care this exception.
as the recognize model converges, the problem disappears

i think i need upload the pretrained icdar2015 model,so you can finetune with it.

**Comment by ongcj** [comment]
> @ongcj
> i think over and over again, i think there is **no nessary** to care this exception.
> as the recognize model converges, the problem disappears
> 
> i think i need upload the pretrained icdar2015 model,so you can finetune with it.

Alright thanks. 
It will be great if you could upload the pretrained ICDAR2015 model. Thanks.

**Comment by novioleo** [comment]
@ongcj ah~

**Comment by ongcj** [comment]
Hi @novioleo, 

im currently still training the model with ICDAR2015 data set and im currently at 400 epoch. I tried running eval.py and the result is very bad. 
TP: 858, FP: 1020, FN: 1373, precision: 0.456869, recall: 0.384581.  

One of the example shown below. 

![image](https://user-images.githubusercontent.com/14144708/59092746-df617b80-8944-11e9-9ec9-a03308f96228.png)


May I know if the poor result is due to some error in my config? Or just that the number of epoch is still too little. The below is my config. 

```
{
    "name": "FOTS",
    "cuda": true,
    "gpus": [0],
    "finetune": "",
    "need_grad_backbone": true,
    "data_loader": {
        "dataset":"icdar2015",
        "data_dir": "./data/OCR/ICDAR2015",
        "batch_size": 4,
        "shuffle": true,
        "workers": 0
    },
    "validation": {
        "validation_split": 0.15,
        "shuffle": true
    },

    "lr_scheduler_type": "ExponentialLR",
    "lr_scheduler_freq": 50,
    "lr_scheduler": {
            "gamma": 0.94
    },

    "optimizer_type": "Adam",
    "optimizer": {
        "lr": 0.0001,
        "weight_decay": 1e-5
    },
    "loss": "FOTSLoss",
    "metrics": ["fots_metric"],
    "trainer": {
        "epochs": 1000,
        "save_dir": "./data/OCR/model/ICDAR2015",
        "save_freq": 1,
        "verbosity": 2,
        "monitor": "loss",
        "monitor_mode": "min"
    },
    "arch": "FOTSModel",
    "model": {
        "mode": "united",
        "scale": 512,
        "crnn": {
            "img_h": 16,
            "hidden": 1024
        },
        "keys": "custom_1"
    }
}

```

**Comment by novioleo** [comment]
@ongcj there is no error in your config,but i don't adapt to icdar2015....so you can't reach the result of paper refered.

**Comment by novioleo** [comment]
can i close this issue now? @ongcj
